parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -228,34 +183,26 @@ pub mod pallet {
 				Error::<T>::PayloadTooLarge,
 			);
 
-			let next_id = <NextId<T>>::get();
-			if next_id.checked_add(1).is_none() {
-				return Err(Error::<T>::Overflow.into())
-			}
+			let nonce = <Nonce<T>>::get(source_id);
+			let next_nonce = nonce.checked_add(1).ok_or(Error::<T>::Overflow)?;
```

Comment:
though not possible in near future, just curious how to recover if `nonce` overflow here, is it better not use  `checked_add` and just let it to be zero?

Reply:
Allowing overflow to nonces exposes us to replay attacks. When the nonce resets to zero, a malicious attacker could resubmit an old message with its valid proofs.

In practice this attack takes a lot of messages over the bridge (2**64 ~= 18.4*10**18), but a known bricking condition we need to upgrade to fix is better than a silently drained bridge.

A lesser concern is that we also lose the ability to sequence all messages for a source, which could make auditing harder. When there are two messages with nonce 0, we can't know which one came first just by looking at the message. We need external information like a timestamp.

Reply:
A permanent fix could be making `Nonce` map to a resizable uint type, instead of `u64`. A wrapper around `&[u64]` might work if there aren't existing libraries.

Reply:
A simpler workaround in the current design is to have sources that approach 2**64 messages switch to a different `sourceId`. Might have knock-on effects though, like the need for a new proxy account on Ethereum. Or others from breaking the assumption of a unique `sourceId` per source.

Reply:
yes, totally make sense.  And 2**64 is big enough the concern here may be unnessessary